shm_wrapper/shm_wrapper.h

```diff
@@ -16,122 +16,225 @@
 //enumerated names for the two associated blocks per device
 typedef enum stream { DATA, COMMAND } stream_t;
 
-// ******************************************* UTILITY FUNCTIONS ****************************************** //
+//shared memory has these parts in it
+typedef struct dev_shm {
```

Comment:
All of these typedefs should stay in `shm.c` since that's the only file that needs to know about them. Only the API functions should exist in this header file

Reply:
Yeah, I was struggling with how to do this, since I didn't want to have to copy-paste this stuff from `shm_wrapper.c` into `shm.c` each time we made any changes to `shm_wrapper.c`, but I guess that's alright. `shm.c` is kind of a special process anyways. I'll leave a note at the top.
👌